# Business Understanding/Overview


I am charged with determining which aircrafts have the lowest risk for the company to start a new business since it is trying to expand and diversify  its portfolio. They are interested in purchasing and operating airplanes for both commercial and private enterprises, but do not know anything about the potential risks of aircraft. My aim is to assist look at the data, analyse  then translate my findings into actionable insights that the head of the new aviation division can use to help decide which aircraft to purchase

# Problem Statement

The company wants to get into  new industries and explore aircrafts for commercial and private purposes and it requires assessment form the data to identify low-risk aircraft. The goal is to recommend low risk aircraft that are suitable for successful market entry. Actionable insights will guide the aviation division in making informed purchasing decisions.


# Objective



1. Identify and compare accident rates versus aircraft model to find the ones with the lowest accident rate and the safest

2. Look at geographical distribution of accidents to identify regions or routes with higher risk which will help in planning and strategic deployment of the aircraft.

3. Analyze past data to identify accident trends over time and determine whether accident rates are improving or worsening.

4. : Compare commercial and private aircraft to understand which would be more profitable and practical for the company to adopt




# Success Criteria



1. The project will be successful if we are able to identify how viable it is to get into the aircraft business through the analysis made

2. If we are  able to deliver clear and analysed accident rates created by specific aircrafts to identify the one with the lowest risk

3. Deliver actionable comparisons between commercial and private aircraft and proof of why either is more profitable



# Limitations and Assumptions




*   Due to missing and incomplete data, we may not get the true picture of the recommendations that would be made
*   We only have Accidents data and that may not be conclusive in datermining whether an aircraft is viable, durable and cost efficient. Safety of the aircraft alone might not be sufficient





# Data Understanding

In [1]:
#Imporrt Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
#Reading the dataset from a csv file using pandas 
#I encountered an error and researched on how to fix it with the below code

df = pd.read_csv('AviationData.csv', encoding='latin1', low_memory=False)


# Aviation Data 

The NTSB aviation accident database contains information from 1962 and later about civil aviation accidents and selected incidents within the United States, its territories and possessions, and in international waters.

In [3]:

df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [4]:
df.tail()

Event.Id Investigation.Type Accident.Number  Event.Date  \
88884  20221227106491           Accident      ERA23LA093  2022-12-26   
88885  20221227106494           Accident      ERA23LA095  2022-12-26   
88886  20221227106497           Accident      WPR23LA075  2022-12-26   
88887  20221227106498           Accident      WPR23LA076  2022-12-26   
88888  20221230106513           Accident      ERA23LA097  2022-12-29   

            Location        Country Latitude Longitude Airport.Code  \
88884  Annapolis, MD  United States      NaN       NaN          NaN   
88885    Hampton, NH  United States      NaN       NaN          NaN   
88886     Payson, AZ  United States  341525N  1112021W          PAN   
88887     Morgan, UT  United States      NaN       NaN          NaN   
88888     Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
88884          NaN  ...          Personal                 NaN   
88885          NaN  ...               NaN                 NaN   
88886       PAYSON  ...          Personal                 NaN   
88887          NaN  ...          Personal  MC CESSNA 210N LLC   
88888          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
88884                  0.0                    1.0                  0.0   
88885                  0.0                    0.0                  0.0   
88886                  0.0                    0.0                  0.0   
88887                  0.0                    0.0                  0.0   
88888                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight Report.Status  \
88884             0.0               NaN                    NaN           NaN   
88885             0.0               NaN                    NaN           NaN   
88886             1.0               VMC                    NaN           NaN   
88887             0.0               NaN                    NaN           NaN   
88888             1.0               NaN                    NaN           NaN   

      Publication.Date  
88884       29-12-2022  
88885              NaN  
88886       27-12-2022  
88887              NaN  
88888       30-12-2022  

[5 rows x 31 columns]

In [5]:
#Checking for shape of the dataset
df.shape

(88889, 31)

In [6]:
df['Investigation.Type']

0        Accident
1        Accident
2        Accident
3        Accident
4        Accident
           ...   
88884    Accident
88885    Accident
88886    Accident
88887    Accident
88888    Accident
Name: Investigation.Type, Length: 88889, dtype: object

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

1. There are 31 columns, including categorical and numerical data, with several columns containing missing values.
2. Most columns are of type object. Columns such as Event.Date, Publication.Date, and Latitude/Longitude may require:
Conversion to appropriate types ie, datetime for date columns, float for geospatial coordinates.

In [8]:
#Checking for any duplicates
df.duplicated().sum()

0

In [9]:
df.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

In [10]:
#Checking for null values
df.isna().sum()

Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     52
Country                     226
Latitude                  54507
Longitude                 54516
Airport.Code              38640
Airport.Name              36099
Injury.Severity            1000
Aircraft.damage            3194
Aircraft.Category         56602
Registration.Number        1317
Make                         63
Model                        92
Amateur.Built               102
Number.of.Engines          6084
Engine.Type                7077
FAR.Description           56866
Schedule                  76307
Purpose.of.flight          6192
Air.carrier               72241
Total.Fatal.Injuries      11401
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Uninjured            5912
Weather.Condition          4492
Broad.phase.of.flight     27165
Report.Status              6381
Publication.Date          13771
dtype: i

In [11]:
# Calculating the missing values
missing_summary = df.isna().sum().sort_values(ascending=False)
missing_percentage = (df.isna().mean() * 100).sort_values(ascending=False)

# Creating and displaying summary DataFrame
missing_data = pd.DataFrame({
    'Missing Count': missing_summary,
    'Missing %': missing_percentage.round(2)
})

# Displaying the columns with the missing values
missing_data = missing_data[missing_data['Missing Count'] > 0]

missing_data

Missing Count  Missing %
Schedule                        76307      85.85
Air.carrier                     72241      81.27
FAR.Description                 56866      63.97
Aircraft.Category               56602      63.68
Longitude                       54516      61.33
Latitude                        54507      61.32
Airport.Code                    38640      43.47
Airport.Name                    36099      40.61
Broad.phase.of.flight           27165      30.56
Publication.Date                13771      15.49
Total.Serious.Injuries          12510      14.07
Total.Minor.Injuries            11933      13.42
Total.Fatal.Injuries            11401      12.83
Engine.Type                      7077       7.96
Report.Status                    6381       7.18
Purpose.of.flight                6192       6.97
Number.of.Engines                6084       6.84
Total.Uninjured                  5912       6.65
Weather.Condition                4492       5.05
Aircraft.damage                  3194       3.59
Registration.Number              1317       1.48
Injury.Severity                  1000       1.12
Country                           226       0.25
Amateur.Built                     102       0.11
Model                              92       0.10
Make                               63       0.07
Location                           52       0.06

# Handling the missing value

The following columns were dropped due to the high percentage of missing values:

- **Schedule (85.85% missing)**:  

- **Air.carrier (81.27% missing)**:  
 
- **FAR.Description (63.97% missing)**:  
  
- **Aircraft.Category (63.68% missing)**:  

- **Longitude and Latitude (61.33% and 61.28% missing, respectively)**:

In [12]:
columns_to_drop = ["Schedule", "Air.carrier", "FAR.Description", "Aircraft.Category", "Longitude", "Latitude"]

# Dropping the columns
df = df.drop(columns=columns_to_drop)

# Verifying the columns have been dropped
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Airport.Code            50249 non-null  object 
 7   Airport.Name            52790 non-null  object 
 8   Injury.Severity         87889 non-null  object 
 9   Aircraft.damage         85695 non-null  object 
 10  Registration.Number     87572 non-null  object 
 11  Make                    88826 non-null  object 
 12  Model                   88797 non-null  object 
 13  Amateur.Built           88787 non-null  object 
 14  Number.of.Engines       82805 non-null

### Columns with missing values that can be inputed

Location (0.06%)

In [13]:
# Checking rows where Location is missing
missing_location_rows = df[df['Location'].isna()]

# Filling missing Location values with Airport.Name
df['Location'] = df['Location'].fillna(df['Airport.Name'])

In [14]:
remaining_missing_location = df['Location'].isna().sum()
remaining_missing_location
remaining_missing_location = df['Location'].isna().sum()
remaining_missing_location

50

We decided to drop the remaining 50 missing location entries because imputing them using the mode could introduce inaccuracies. Also, 50 represents only a small percentage of the dataset.

In [15]:
# Dropping rows where Location is missing
df = df.dropna(subset=['Location'])

In [16]:
# Verifying if all missing values in 'Location' are handled
remaining_missing_location = df['Location'].isna().sum()
remaining_missing_location

0

## Data Preparation


In [17]:
#Take a look at the columns
print(df.columns)


Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Airport.Code', 'Airport.Name',
       'Injury.Severity', 'Aircraft.damage', 'Registration.Number', 'Make',
       'Model', 'Amateur.Built', 'Number.of.Engines', 'Engine.Type',
       'Purpose.of.flight', 'Total.Fatal.Injuries', 'Total.Serious.Injuries',
       'Total.Minor.Injuries', 'Total.Uninjured', 'Weather.Condition',
       'Broad.phase.of.flight', 'Report.Status', 'Publication.Date'],
      dtype='object')


In [18]:
#Drop the columns that will not be used, some are same , others have missing values and others wont be necessary for the analysis
df = df.drop(columns=['Investigation.Type','Country','Accident.Number', 'Latitude', 'Longitude', 'Airport.Code',
                      'Airport.Name','FAR.Description','Schedule','Air.carrier', 'Broad.phase.of.flight', 'Report.Status' ])

KeyError: "['Latitude' 'Longitude' 'FAR.Description' 'Schedule' 'Air.carrier'] not found in axis"

In [ ]:
df

In [ ]:
#Identify and compare accident rates versus aircraft model to find the ones with the lowest accident rate and the safest

In [ ]:
#Retrieving the different aircraft models available
df['Model']

In [ ]:
#Retrieving the unique values from the 'Model'
df['Model'].unique()

In [ ]:
#Getting to know the count of unique values there are each model 
df['Model'].value_counts()

In [ ]:
# Get the top 10  most common models
top_models = df['Model'].value_counts().head(10)
top_models

In [ ]:
#Safety Analysis: Focus on analyzing the safety records of the most common models.

In [ ]:

# Create a bar chart
plt.figure(figsize=(12, 6))
top_models.plot(kind='bar', color='skyblue')
plt.title('Top 10 Most Common Aircraft Models in the Dataset')
plt.xlabel('Aircraft Model')
plt.ylabel('Number of Accidents')
plt.xticks(rotation=45)
plt.tight_layout()
plt.show()
